### ISSUE TITLE
Would you be interested in publishing klipse to bower

### ISSUE THREAD

**Issue by dvberkel** [viebel/klipse#121]
First of all: I really love what you created. Great idea very well executed!

Unless I am mistaken, if you want to use klipse, you fetch it from `http://app.klipse.tech/plugin_prod/js/`. This means you always need a internet connection if you want to use it. Sometimes I do not want to rely on internet.

Obviously I could manually download klipse and store it locally and reference it from there. I would rather do this in an automated fashion. I personally would like to use bower for that.

[Bower](https://bower.io/) calls itself a package manager for the web. It allows to specify which dependencies your project has and fetch them with the bower command line interface.

Would you be interested in publishing your artifacts to bower?

To have a feel for what you get yourself into below your find a series of steps

1. Keep the build artifacts in the repository.
2. Create a `bower.json` describing the project and what to serve.
3. Register klipse to bower. This needs to be done only once.
4. Tag each new release.

I am willing and able to put in the  work of creating `bower.json` in a pull-request.

**Comment by viebel** [comment]
Sounds like a good idea!

Please go ahead and prepare the pull-request

Why not `npm` also?

What is your main language of interest with Klipse?

Are you planning on using klipse in your docs/slides?

**Comment by dvberkel** [comment]
My use case is interactive code for slides.

I will prepare a pull-request, probably by tomorrow.

What would the use case for `npm` be? If you are interested I could do a separate pull request for `npm`

**Comment by viebel** [comment]
`npm` serves the same use case as `bower`. 
I guess that some people might prefer one over the other.

Did you see the slides I made at http://slides.klipse.tech?

I tried a `reveal.js` but it was a mess with CodeMirror. Therefore I switched to `deck.js`.

That language are you going to target?

**Comment by dvberkel** [comment]
I just watched your slides. You bring home the point of KLIPSE very well.

I intended to use `reveal.js`. What problems did you experience. I am guessing styling was an issue?

The target language for my first use is clojure(script).

**Comment by viebel** [comment]
CSS conflicts with code mirror. See
https://github.com/codemirror/CodeMirror/issues/4189

I gave a talk at Clojutre 2016 - with those slides:
https://www.youtube.com/watch?v=33mTFjPIN10&feature=youtu.be

**Comment by titaniumbones** [comment]
@dvberkel, have you made any progress on this? I would like to try this too, preferably with integration with org-mode and org-reveal.  Will help if my limited skills permit.

**Comment by viebel** [comment]
@titaniumbones klipse is available in bower

```
bower install klipse
```

I have been able to itegrate klipse with `deck.js` as you can see here https://github.com/viebel/slides.klipse.tech/blob/master/templates/deckjs/index.html

I had css issues between codemirror and reveal https://github.com/codemirror/CodeMirror/issues/4189

Regarding bower, I really recommend to use the **live** version of klipse instead of the locally saved one. It will ensure you to get fresh updates.

**Comment by titaniumbones** [comment]
Thanks @viebel. I'll try iwth deck., I just wondered if @dvberkel had worked on the reveal integration (only because all my slides are written for reveal, and it will be a little frustrating to have to rework them). Thanks in any case, I'll try figure out dek.

**Comment by viebel** [comment]
@titaniumbones may I ask what are your slides about? What language(s)? Who is the audience?

**Comment by dvberkel** [comment]
@titaniumbones I haven't worked on the preparing klipse for reveal. I know that it is notoriously hard because reveal has very invasive css. If we really want to take it up we probably should open a new issue

**Comment by titaniumbones** [comment]
Maybe a new issue in the reveal repo @dvberkel?

Daan van Berkel writes:

> @titaniumbones I haven't worked on the preparing klipse for 
> reveal. I know that it is notoriously hard because reveal has 
> very invasive css. If we really want to take it up we probably 
> should open a new issue
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the 
> thread.


--

**Comment by dvberkel** [comment]
@titaniumbones It probably is harder to change the way reveal works than to try to integrate klipse into reveal. Furthermore reveal is not invested in making klipse work with reveal, but klipse is. Or at least so it seems

Having said that, it is a good thing to cross reference to reveal, because this is a problem that crops up more often.

**Comment by dvberkel** [comment]
I created issue #133 so that we can continue the discussion there
